great_expectations/checkpoint/checkpoint.py

```diff
@@ -357,7 +357,7 @@ def get_substituted_config(
         if runtime_kwargs is None:
             runtime_kwargs = {}
 
-        config_kwargs: dict = self.get_config(mode=ConfigOutputModes.JSON_DICT)
+        config_kwargs: dict = self.get_config(mode=ConfigOutputModes.JSON_DICT)  # type: ignore[assignment] # always returns a dict
```

Comment:
If it always return a dict, do we need this? Does the explicit annotation cause an issue? Should we cast instead?

Reply:
I don't want to cast because that's like ignoring all errors.

I want to make `ConfigPeer`  a generic and define `overloads` for `.get_config()` but that won't work until `ConfigPeer` is a generic.

I can add a comment explaining these aspirational goals.

Reply:
It is not clear to me either why the `ignore` statement is needed; hence, an explanation would be useful.  Thanks.

Reply:
because `get_config()` is a union of `dict` `BaseConfigYaml` and `str`